lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/errors/ExceptionTranslator.java

```diff
@@ -0,0 +1,131 @@
+package de.tum.in.www1.artemis.lecture.web.rest.errors;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageConversionException;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.problem.DefaultProblem;
+import org.zalando.problem.Problem;
+import org.zalando.problem.ProblemBuilder;
+import org.zalando.problem.StatusType;
+import org.zalando.problem.spring.web.advice.ProblemHandling;
+import org.zalando.problem.spring.web.advice.security.SecurityAdviceTrait;
+import org.zalando.problem.violations.ConstraintViolationProblem;
+
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.FieldErrorVM;
+import tech.jhipster.config.JHipsterConstants;
+import tech.jhipster.web.util.HeaderUtil;
+
+/**
+ * Controller advice to translate the server side exceptions to client-friendly json structures.
+ * The error response follows RFC7807 - Problem Details for HTTP APIs (https://tools.ietf.org/html/rfc7807).
+ */
+@ControllerAdvice
```

Comment:
Is this something we should keep in the specific microservice?

Reply:
I think so, every microservice will define exceptions that are specific for it, i.e. the user management defines exceptions for the cases when the user who tries to register uses an email which is already taken which is not related to other microservices

Reply:
The thing here is that I don't see any microservice specific functionality in this translator. Sure, every MS defines their own exceptions but they are not specifically called here.

Reply:
Ah, now I understood. It actually translates BadRequestAlertException and will later translate EntityNotFound which we import from Artemis because they are common for all microservices. We do not reuse the ExceptioTranslator from Artemis because there are more exceptions that we need to import which are not needed from the service.